src/perl/Bio/EnsEMBL/Pipeline/Runnable/EG/LoadGFF3/LoadGFF3.pm

```diff
@@ -398,20 +400,78 @@ sub add_pseudogenic_transcript {
   return $transcript;
 }
 
+sub update_gene_coords_based_on_transcripts_on_circular_sr {
+  my ($self, $gene, $ga) = @_;
+
+  # fixing only on circular seq_regions
+  return if (!$gene->slice->is_circular);
+  
+  # we can assume that we work with the top-level seq_regions,
+  #   thus Gene::start/end and Gene::seq_region_start/end are equivalent
+  my ($gene_start, $gene_end) = ( $gene->start, $gene->end );
+  $gene->recalculate_coordinates();
+  if ($gene_start != $gene->start || $gene_end != $gene->end) {
+    # ideally, just call
+    #   $ga->update_coords($gene);
+    # unfortunately this doesn't work for circular seq_regions
+    # ( Gene::seq_region_start/end uses BaseFeatureAdaptor::_seq_region_boundary_from_db failback for circular regions )
+    # going vanila SQL update way
+    $self->update_gene_seq_region_coords_from_coords($gene, $ga);
+  }
+}
+
+sub update_gene_seq_region_coords_from_coords {
```

Comment:
It's not clear what the circular case does differently from the default update_coords? Please add documentation

Reply:
See on L414. should I copy it?

Reply:
https://github.com/Ensembl/ensembl/blob/release/111/modules/Bio/EnsEMBL/Feature.pm#L1067 -- for me

Reply:
added